### PR TITLE
Updating development docs with information on prow.

### PR DESCRIPTION
**Pull request by mohammedzee1000**
Signed-off-by: Mohammed Zeeshan Ahmed <mohammed.zee1000@gmail.com>

## What is the purpose of this change? What does it change?

Updating our development documentation so that it has necessary information about the prow workflow

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -90,12 +90,21 @@ Terminology we use:
...
   - Does this feature / fix work locally for me? 
   - Am I able to understand the code correctly / have comments been added to the code?
 
+#### Prow best practices
+
+  - Get used to how prow works and its commands. 
+    - Prow uses [OWNERS](https://github.com/kubernetes/community/blob/master/contributors/guide/owners.md) files to determine who can approve and lgtm a PR.
+    - List of possible bot [commands](https://prow.k8s.io/command-help)
+  - As far as possible, do not `/lgtm` a PR, until it is mergable (we need atleast 2 approving reviews and no request changes left and pre-submit tests should pass) or if you do, put it on hold with `/hold` immediately. Either of them can be cancelled as `/lgtm cancel` or `/hold cancel`. This especially applies to **draft PRs**
+  - Approvers can use `/approve` and `/approve cancel` respectively to give their approval. **Note:** approvers github approval also counts the same as `/approve`
```
Comment (kadel) [inline docs/development.md]: Can you please add short explanation how are approves (approvers) different from lgtm (reviewers)?

Comment (mohammedzee1000) [reply docs/development.md]: Sure

Comment (mohammedzee1000) [reply docs/development.md]: Added

File: docs/development.md [hunk 2]
```diff
@@ -90,12 +90,24 @@ Terminology we use:
...
 
   - Have tests been added?
   - Does this feature / fix work locally for me? 
   - Am I able to understand the code correctly / have comments been added to the code?
 
+#### Prow best practices
+
+  - Get used to how prow works and its commands. 
+    - Prow uses [OWNERS](https://github.com/kubernetes/community/blob/master/contributors/guide/owners.md) files to determine who can approve and lgtm a PR.
+    - List of possible bot [commands](https://prow.k8s.io/command-help)
```
Comment (stevekuznetsov) [inline docs/development.md]: Link to https://deck-ci.svc.ci.openshift.org/command-help, that is the wrong Prow

Comment (kadel) [reply docs/development.md]: Thank you for checking.
fixing in https://github.com/redhat-developer/odo/pull/1438

**Pull request merged**